core/modules/ccontrol/testCControl.cc

```diff
@@ -686,6 +707,246 @@ static const std::vector<Antlr4CompareQueries> ANTLR4_COMPARE_QUERIES = {
+            query::SelectList& selectList = selectStatement->getSelectList();
+            (*selectList.getValueExprList())[1]->getFactorOpsRef()[0].op = query::ValueExpr::MOD;
+            auto whereClauseRef = selectStatement->getWhereClause();
+            auto orTerm = std::dynamic_pointer_cast<query::OrTerm>(whereClauseRef.getRootTerm());
+            auto andTerm = std::dynamic_pointer_cast<query::AndTerm>(orTerm->_terms[0]);
+            auto boolFactor = std::dynamic_pointer_cast<query::BoolFactor>(andTerm->_terms[0]);
+            auto compPredicate = std::dynamic_pointer_cast<query::CompPredicate>(boolFactor->_terms[0]);
+            query::ValueExpr::FactorOp& factorOp = compPredicate->left->getFactorOpsRef()[0];
+            factorOp.op = query::ValueExpr::MOD;
+        },
+        "SELECT objectId,(ra_PS MOD 3),decl_PS FROM Object WHERE (ra_PS MOD 3)>1.5"
+    ),
+
+    // tests the DIV operator
+    Antlr4CompareQueries(
+        "SELECT objectId from Object where ra_PS DIV 2 > 1",
+        "SELECT objectId from Object where ra_PS/2 > 1",
+        [](query::SelectStmt::Ptr const & selectStatement) {
+              auto orTerm = std::dynamic_pointer_cast<query::OrTerm>(selectStatement->getWhereClause().getRootTerm());
+              auto andTerm = std::dynamic_pointer_cast<query::AndTerm>(orTerm->_terms[0]);
+              auto boolFactor = std::dynamic_pointer_cast<query::BoolFactor>(andTerm->_terms[0]);
+              auto compPredicate = std::dynamic_pointer_cast<query::CompPredicate>(boolFactor->_terms[0]);
+              query::ValueExpr::FactorOp& factorOp = compPredicate->left->getFactorOpsRef()[0];
+              factorOp.op = query::ValueExpr::DIV;
+        },
+        "SELECT objectId FROM Object WHERE (ra_PS DIV 2)>1"
+    ),
+
+    // tests the & operator
+    Antlr4CompareQueries(
+        "SELECT objectId from Object where objectID & 1 = 1",
+        "SELECT objectId from Object where objectID = 1",
+        [](query::SelectStmt::Ptr const & selectStatement) {
+              auto orTerm = std::dynamic_pointer_cast<query::OrTerm>(selectStatement->getWhereClause().getRootTerm());
+              auto andTerm = std::dynamic_pointer_cast<query::AndTerm>(orTerm->_terms[0]);
+              auto boolFactor = std::dynamic_pointer_cast<query::BoolFactor>(andTerm->_terms[0]);
+              auto compPredicate = std::dynamic_pointer_cast<query::CompPredicate>(boolFactor->_terms[0]);
+              compPredicate->left = std::make_shared<query::ValueExpr>(query::ValueExpr::FactorOpVector({
+                  query::ValueExpr::FactorOp(
+                          query::ValueFactor::newColumnRefFactor(query::ColumnRef::newShared("", "", "objectID")),
+                          query::ValueExpr::BIT_AND),
+                  query::ValueExpr::FactorOp(query::ValueFactor::newConstFactor("1"))
+              }));
+        },
+        "SELECT objectId FROM Object WHERE (objectID&1)=1"
```

Comment:
qserv serializes the IR without spaces, this is the same behavior as before I started working on the antlr4 parser. For now I'm going to leave it as is. If we want to address it we can do it in separate work.

Reply:
qserv serializes the intermediate representation without spaces, it's been this way since before I started working on the new parser. If we want to change this behavior it should be done as separate work.

These queries are not executed, this just tests for expected IR representation & serialization. So the column name need not match any of the 'normal' qserv column naming conventions, e.g. `objectId` vs `objectID`